components/compositing/compositor.rs

```diff
@@ -1539,51 +1542,83 @@ impl<Window: WindowMethods + ?Sized> IOCompositor<Window> {
 
     fn on_touch_down(&mut self, identifier: TouchId, point: DevicePoint) {
         self.touch_handler.on_touch_down(identifier, point);
-        self.send_touch_event(TouchEventType::Down, identifier, point);
+        self.send_touch_event(
+            TouchEventType::Down,
+            identifier,
+            point,
+            TouchAction::NoAction,
+        );
     }
 
     fn on_touch_move(&mut self, identifier: TouchId, point: DevicePoint) {
-        match self.touch_handler.on_touch_move(identifier, point) {
-            TouchAction::Scroll(delta) => self.on_scroll_window_event(
-                ScrollLocation::Delta(LayoutVector2D::from_untyped(delta.to_untyped())),
-                point.cast(),
-            ),
-            TouchAction::Zoom(magnification, scroll_delta) => {
-                let cursor = Point2D::new(-1, -1); // Make sure this hits the base layer.
-
-                // The order of these events doesn't matter, because zoom is handled by
-                // a root display list and the scroll event here is handled by the scroll
-                // applied to the content display list.
-                self.pending_scroll_zoom_events
-                    .push(ScrollZoomEvent::PinchZoom(magnification));
-                self.pending_scroll_zoom_events
-                    .push(ScrollZoomEvent::Scroll(ScrollEvent {
-                        scroll_location: ScrollLocation::Delta(LayoutVector2D::from_untyped(
-                            scroll_delta.to_untyped(),
-                        )),
-                        cursor,
-                        event_count: 1,
-                    }));
-            },
-            TouchAction::DispatchEvent => {
-                self.send_touch_event(TouchEventType::Move, identifier, point);
-            },
-            _ => {},
+        let action = self.touch_handler.on_touch_move(identifier, point);
+        if let TouchAction::NoAction = action {
+        } else {
+            self.send_touch_event(TouchEventType::Move, identifier, point, action);
```

Comment:
```suggestion
        if let TouchAction::NoAction != action {
                    self.send_touch_event(TouchEventType::Move, identifier, point, action);
```

You may need to add an `derive` for `PartialEq` for `TouchAction`.

Reply:
Changed, thank you